algorithms/src/snark/marlin/data_structures/proof.rs

```
@@ -26,6 +30,8 @@ use snarkvm_utilities::{
     ToBytes,
 };
 
+use std::collections::BTreeMap;
+
 #[derive(Clone, Debug, PartialEq, Eq, CanonicalSerialize, CanonicalDeserialize)]
 pub struct Commitments<E: PairingEngine> {
     pub witness_commitments: Vec<WitnessCommitments<E>>,
```

Let's change this to `witness_commitments: Vec<Vec<WitnessCommitments<E>>>`, i.e. one inner `Vec` per circuit.

---

As discussed elsewhere:

pros: 
- we would win a static assertion that witness_commitments.len() == g_a_commitments.len())
- It would reflect the structure of z_b_evals though, which is a nice symmetry.

cons:
- we already have checks for length consistency in place for related objects and we'd fail loudly if we mess up the order in witness_commitments
- witness_commitments is created and consumed as flat vector

Without strong benefit I prefer to leave it as is to save time for other bugs/features